include/qpid/dispatch/iterator.h

```diff
@@ -180,6 +180,8 @@ void qd_address_iterator_set_phase(qd_field_iterator_t *iter, char phase);
  */
 void qd_address_iterator_override_prefix(qd_field_iterator_t *iter, char prefix);
 
+void qd_address_iterator_set_length(qd_field_iterator_t *iter, int length);
```

Comment:
Two things:  This function should be a "qd_field_iterator..." function, not a "qd_address_iterator..." function.  There is nothing address-specific about this operation.

Also, it might be clearer to call this "qd_field_iterator_trim" with an argument called "new_length".  This suggests that the length can only be reduced.